TACC/Core-Portal
# Frontend builds using Vite

## Overview: ##

![hmr_vite](https://user-images.githubusercontent.com/12601812/136137629-017f5324-8628-46cf-96ff-bf6d2f55817c.gif)

Draft PR for migration to the Vite build tool (better HMR, faster builds, no webpack config)

- [x] Rename component files from .js to .jsx
- [x] Replace `styleName` with `import styles from ***.module.css`
- [x] Version bump react to v17 (needed for fast HMR)
- [x] Fix linting: Because of dependency changes I'm not able to exactly reproduce our old linting setup. I propose that we separate our linting concerns between functional warnings (unused imports, accessibility) and styling (whitespace, semicolons, trailing commas). ESLint should handle the former and Prettier should handle the latter.
- [x] Switch between dev server and production bundle based on Django DEBUG setting

## Summary of Changes: ##

## Testing Steps (local): ##
1. `npm ci && npm run dev`

## Testing Steps (production build): ##
1. Stop all running containers.
2. `npm ci && npm run build`
3. set `_DEBUG = False` in `settings_custom.py`
4. Tweak the nginx conf to prevent Django from trying to serve static files. In `server/conf/nginx/nginx.conf` line 83 should read:
> location ~ ^/(auth|workbench|tickets|googledrive-privacy-policy|public-data|accounts|api|login|webhooks|search) {
5. Restart all containers.
5. Shell into the Django container (`docker exec -it core_portal_django /bin/bash`) and run `python manage.py collectstatic`
6. Navigate to the the dev portal and check that the UI renders correctly. In your network tab, the index and vendor bundle hashes should match the ones in `client/dist/`: 
![image](https://user-images.githubusercontent.com/12601812/138536825-191b9050-cfc7-45e5-ae71-fca836ba5b46.png)


## UI Photos:

## Notes: ##
